### ISSUE TITLE
Error: USER is set to "xx" but we expect "yy"

### ISSUE THREAD

**Issue by lamarios** [ublue-os/fleek#205]
**Describe the bug**
Everything use to work fine, but suddenly after fleek update then fleek apply I have the following error:
```
Error: USER is set to "xx" but we expect "yy"
```

I have multiple users on this machine and this happens when i switch user. I can manually fix it if i edit the file in   ~/.local/share/fleek/my-hostname/my-hostname.nix

and replace with the currently logged in user

**To Reproduce**
Have multiple users on the same machine. Fleek update / fleek apply using both users

**Expected behavior**
Fleek apply shouldn't throw errors about expected users

**Environment**
 - OS: fedora silverblue
 - Shell: zsh
 - fleek version 0.9.3

**Additional context**
```fleek apply     
  [d]  debug enabled
 [d]  go	go1.19.6
      path	github.com/ublue-os/fleek/cmd/fleek
      mod	github.com/ublue-os/fleek	(devel)	
      dep	atomicgo.dev/cursor	v0.1.1	
      dep	atomicgo.dev/keyboard	v0.2.9	
      dep	github.com/ProtonMail/go-crypto	v0.0.0-20230217124315-7d5c6f04bbb8	
      dep	github.com/atotto/clipboard	v0.1.4	
      dep	github.com/aymanbagabas/go-osc52/v2	v2.0.1	
      dep	github.com/charmbracelet/bubbles	v0.14.0	
      dep	github.com/charmbracelet/bubbletea	v0.23.1	
      dep	github.com/charmbracelet/lipgloss	v0.7.1	
      dep	github.com/cloudflare/circl	v1.1.0	
      dep	github.com/containerd/console	v1.0.3	
      dep	github.com/emirpasic/gods	v1.18.1	
      dep	github.com/erikgeiser/promptkit	v0.8.0	
      dep	github.com/fatih/color	v1.15.0	
      dep	github.com/fitv/go-i18n	v1.0.4	
      dep	github.com/go-git/gcfg	v1.5.0	
      dep	github.com/go-git/go-billy/v5	v5.4.1	
      dep	github.com/go-git/go-git/v5	v5.6.1	
      dep	github.com/google/uuid	v1.3.0	
      dep	github.com/gookit/color	v1.5.3	
      dep	github.com/hashicorp/go-version	v1.6.0	
      dep	github.com/imdario/mergo	v0.3.13	
      dep	github.com/jbenet/go-context	v0.0.0-20150711004518-d14ea06fba99	
      dep	github.com/kevinburke/ssh_config	v1.2.0	
      dep	github.com/lithammer/fuzzysearch	v1.1.5	
      dep	github.com/lucasb-eyer/go-colorful	v1.2.0	
      dep	github.com/mattn/go-colorable	v0.1.13	
      dep	github.com/mattn/go-isatty	v0.0.17	
      dep	github.com/mattn/go-localereader	v0.0.1	
      dep	github.com/mattn/go-runewidth	v0.0.14	
      dep	github.com/muesli/ansi	v0.0.0-20221106050444-61f0cd9a192a	
      dep	github.com/muesli/cancelreader	v0.2.2	
      dep	github.com/muesli/mango	v0.1.0	
      dep	github.com/muesli/mango-cobra	v1.2.0	
      dep	github.com/muesli/mango-pflag	v0.1.0	
      dep	github.com/muesli/reflow	v0.3.0	
      dep	github.com/muesli/roff	v0.1.0	
      dep	github.com/muesli/termenv	v0.15.1	
      dep	github.com/otiai10/copy	v1.11.0	
      dep	github.com/pjbgf/sha1cd	v0.3.0	
      dep	github.com/pkg/errors	v0.9.1	
      dep	github.com/pterm/pterm	v0.12.59	
      dep	github.com/rivo/uniseg	v0.4.4	
      dep	github.com/riywo/loginshell	v0.0.0-20200815045211-7d26008be1ab	
      dep	github.com/sergi/go-diff	v1.2.0	
      dep	github.com/skeema/knownhosts	v1.1.0	
      dep	github.com/spf13/cobra	v1.7.0	
      dep	github.com/spf13/pflag	v1.0.5	
      dep	github.com/xanzy/ssh-agent	v0.3.3	
      dep	github.com/xo/terminfo	v0.0.0-20220910002029-abceb7e1c41e	
      dep	golang.org/x/crypto	v0.6.0	
      dep	golang.org/x/net	v0.7.0	
      dep	golang.org/x/sys	v0.7.0	
      dep	golang.org/x/term	v0.7.0	
      dep	golang.org/x/text	v0.9.0	
      dep	gopkg.in/warnings.v0	v0.1.2	
      dep	gopkg.in/yaml.v3	v3.0.1	
      build	-compiler=gc
      build	-tags=netgo
      build	-trimpath=true
      build	CGO_ENABLED=0
      build	GOARCH=amd64
      build	GOOS=linux
      build	GOAMD64=v1
 [d]  installed at: /nix/store/p1ynls2ky4jjbc52c72py872r2vvhmkb-fleek-v0.9.3/bin
 [d]  git autopush: true
 [d]  git autocommit: true
 [d]  git autopull: true
 [*]  Apply fleek configuration
 [d]  is git repo
 [d]  git will pull
From github.com:lamarios/fleek
 * branch            main       -> FETCH_HEAD
Already up to date.
 [✓]  Writing configuration files                                                                                                                                  
 [d]  is git repo
 [d]  git will add
 [d]  git will commit
 [d]  git status is empty, skipping commit
 [d]  git will push
Everything up-to-date
 [i]  Applying configuration
Starting Home Manager activation
Error: USER is set to "xx" but we expect "yy"
  ERROR   
  ERROR   Error: exit status 1
  ERROR   
 [d]  Command stderr: 

 [d]  
      ExecutionID:d26b8c60bc6d4e6cb8a85cca9401120a

```

**Comment by bketelsen** [comment]
fleek does not support multiple users on the same machine. It's a limitation of the underlying home-manager implementation.

**Comment by bketelsen** [comment]
thinking through this a little more it might be possible to support this. I'll experiment!

**Comment by lamarios** [comment]
It should after digging a bit more into understanding home-manager, i made my own config where it is possible to have as the flake config names are arbitrary

**Comment by bketelsen** [comment]
there's a fix for this in #214
